src/ApplicationInsights.Kubernetes/K8sInfoBackgroundService.cs

```diff
@@ -30,6 +30,26 @@ internal class K8sInfoBackgroundService : BackgroundService
         _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
     }
 
+    // Allows non-hosted service to bootstrap cluster info fetch.
+    void IK8sInfoBootstrap.Run()
```

Comment:
Can we make this idempotent so it is less error-prone? I.e.
- OK for user code to call this multiple times (the cluster info is guaranteed to be initialized only once)
- OK to call it even if service host is present

This should be relatively easy to achieve using `Lazy<T>` or `LazyInitializer`.

Reply:
Exactly what I was thinking too. The only reason I didn't put it out was that I am worried that the current implementation might not be the most optimal one, and I want to first release a beta package for the customer to try and see if they like the signature or not before working on improving the details.
I think I can have another PR to improve the implementations, what do you think?

Reply:
It turned out to be a bit more complex than that. For Lazy<Task>, it takes more than just that to have a cancellation token. An alternative that I see is to inherit from `Lazy<T>` to form a `Lazy<Task>`, having cancellation token support there, it is more effort than this bug fix. I will file another issue to track it.